retrofit/src/main/java/retrofit2/RequestFactory.java

```diff
@@ -98,6 +101,10 @@ static RequestFactory parseAnnotations(Retrofit retrofit, Method method) {
           + ") doesn't match expected count (" + handlers.length + ")");
     }
 
+    if (isKotlinSuspendFunction) {
+      // The Continuation is the last parameter and the handlers array contains null at that index.
```

Comment:
Wondering aloud: could you symmetrically synthesize the return type from this argument? It’s how the coroutine and non-coroutine flows could fit together.

```
interface ResponseReturner<>T {
  void success (Response<T> response);
  void failure (Throwable t);
}
```

Reply:
I guess where I'm going is to imagine coroutines is actually a more general case and we can model regular returns within the coroutine machinery. Though this would only be for API symmetry.

Reply:
What you're describing sounds like the callback factory that I originally modeled. It seems like a lot of work for something I expect no one except coroutines to use.

Reply:
Works for me! None of it leaks into the public API anyway so it doesn’t much matter.

One other way of potentially looking at this:

The coroutines support synthesizes a return type of `Call<T>` or `Call<Response<T>>` and just calls `await()` for you